src/Sentry/Internal/Hub.cs

```diff
@@ -258,6 +260,7 @@ public void Dispose()
                 }
             }
 
+            Interlocked.Exchange(ref _isEnabled, 0);
```

Comment:
Does this helps us in any way? I'm thinking because by this point we're already half way into disposing anyway, so if there was a race condition, it could have occurred already. Maybe just an `bool _isDisposed` field is enough? Which scenarios are we trying to protect against here?

Reply:
Tbh, this is what we discussed with Bruno. My initial thought was the same, just `bool`, but Bruno insisted on `Interlocked`. From my understanding, the main scenario for this is that `Dispose` can be called by more than 1 thread.

Reply:
> but Bruno insisted on Interlocked

I don't recall a conversation where I insisted. 
What I recall is that I suggested a solution with `Interlocked` and there was no discussion of trade off.